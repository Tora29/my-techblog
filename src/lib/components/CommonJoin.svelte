<script lang="ts">
	// コンポーネントの変数
	export let joinItems: { link: string; class: string; text: string }[];

	// コンポーネントのインポート
	import CommonButton from './CommonButton.svelte';
</script>

<div class="join">
	{#each joinItems as item}
		<CommonButton link={item.link} classType={item.class} text={item.text} />
	{/each}
</div>
