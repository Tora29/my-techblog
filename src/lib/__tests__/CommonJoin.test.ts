import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import CommonJoin from '$lib/components/CommonJoin.svelte';

describe('Join Component', () => {
	test('ジョインをデフォルトプロパティでレンダリング', () => {
		// テスト用デフォルトデータの準備
		const joinItems = [
			{
				link: '/about',
				class: 'btn-primary',
				text: 'ABOUT ME'
			},
			{
				link: '/blog',
				class: 'btn-primary',
				text: 'BLOG'
			},
			{
				link: '/service',
				class: 'btn-primary',
				text: 'SERVICE'
			}
		];

		// コンポーネントをレンダリングし、joinItems をprops として渡す
		const { getByText, container } = render(CommonJoin, {
			props: { joinItems }
		});

		// ボタンがレンダリングされるか確認
		const aboutButton = getByText('ABOUT ME');
		const blogButton = getByText('BLOG');
		const serviceButton = getByText('SERVICE');

		// ボタンの存在確認
		expect(aboutButton).toBeInTheDocument();
		expect(blogButton).toBeInTheDocument();
		expect(serviceButton).toBeInTheDocument();

		// ボタンのクラス確認
		expect(aboutButton).toHaveClass('btn-primary');
		expect(blogButton).toHaveClass('btn-primary');
		expect(serviceButton).toHaveClass('btn-primary');

		// ボタンのリンク確認
		expect(aboutButton.closest('a')).toHaveAttribute('href', '/about');
		expect(blogButton.closest('a')).toHaveAttribute('href', '/blog');
		expect(serviceButton.closest('a')).toHaveAttribute('href', '/service');

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
