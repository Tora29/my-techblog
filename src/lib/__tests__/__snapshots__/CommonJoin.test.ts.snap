// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Join Component > ジョインをデフォルトプロパティでレンダリング 1`] = `
<div>
  <div
    class="join"
  >
    <a
      class="btn btn-primary "
      href="/about"
    >
      ABOUT ME
    </a>
    <a
      class="btn btn-primary "
      href="/blog"
    >
      BLOG
    </a>
    <a
      class="btn btn-primary "
      href="/service"
    >
      SERVICE
    </a>
  </div>
</div>
`;
